src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/OverhangFixingManager.java

```diff
@@ -251,51 +285,95 @@ protected static boolean isRightOverhang(final GenomeLoc readLoc, final GenomeLo
                                                final byte[] reference,
                                                final int referenceStartIndex,
                                                final int spanToTest) {
         // don't process too small a span, too large a span, or a span that is most of a read
-        if ( spanToTest < 1 || spanToTest > MAX_BASES_IN_OVERHANG || spanToTest > read.length / 2 ) {
+        if ( spanToTest < 1 || spanToTest > maxBasesInOverhang || spanToTest > readLength / 2 ) {
             return false;
         }
 
         int numMismatchesSeen = 0;
         for ( int i = 0; i < spanToTest; i++ ) {
             if ( read[readStartIndex + i] != reference[referenceStartIndex + i] ) {
-                if ( ++numMismatchesSeen > MAX_MISMATCHES_IN_OVERHANG ) {
+                if ( ++numMismatchesSeen > maxMismatchesInOverhang) {
                     return true;
                 }
             }
         }
 
         // we can still mismatch overall if at least half of the bases mismatch
+
         return numMismatchesSeen >= ((spanToTest+1)/2);
     }
 
     /**
      * Close out the manager stream by clearing the read cache
      */
     public void close() {
+        writeReads(0);
+    }
+
+    /**
+     * Writes read groups off the top of waitingReads until the total number of reads in the set is less than the
+     * target queue size. If outputToFile == false, then the it will instead mark the first item to setMateChanged in
+     * mateChagnedReads
+     */
+    private void writeReads(int targetQueueSize) {
         // write out all of the remaining reads
-        while ( ! waitingReads.isEmpty() ) {
-            writer.addRead(waitingReads.poll().read);
+        while ( getNReadsInQueue() > targetQueueSize ) {
+            List<SplitRead> waitingGroup = waitingReadGroups.poll();
+            waitingReads = waitingReads - waitingGroup.size();
+
+            // Repair the supplementary groups together and add them into the writer
+            if (outputToFile) {
+                SplitNCigarReads.repairSuplementaryTags(waitingGroup.stream()
+                        .map( r -> r.read )
+                        .collect(Collectors.toList()), header);
+                for (SplitRead splitRead : waitingGroup) {
+                    writer.addRead(splitRead.read);
+                }
+
+                // On the first traversal we want to store reads that would be the mate
+            } else {
+                // Don't mark the readgroup if it is secondary (mate information should always point to the primary alignment)
+                if (!waitingGroup.get(0).read.isSecondaryAlignment() && waitingGroup.get(0).hasBeenOverhangClipped()) {
+                    waitingGroup.get(0).setMateChanged();
+                }
+            }
         }
     }
 
+    // Sets the tool to start writing its output to the file writer
+    public void activateWriting() {
+        close();
```

Comment:
Could you log how many changed reads are stored at the time of activation?  

Also, are you allowed to call activateWriting multiple times?  Should that be an error condition or is there an expected usage that you might want to do that?